stable/cluster-overprovisioner/templates/additionalDeployments.yaml

```diff
@@ -0,0 +1,53 @@
+{{- $relName := .Release.Name -}}
+{{- $relService := .Release.Service -}}
+{{- $fullname :=  include "cluster-overprovisioner.fullname" . -}}
+{{- $name :=  include "cluster-overprovisioner.name" . -}}
+{{- $chart :=  include "cluster-overprovisioner.chart" . -}}
+{{- $chartName := .Chart.Name }}
+{{- $priorityClassName := .Values.priorityClassOverprovision.name }}
+{{- $repository := .Values.image.repository }}
+{{- $imageTag := .Values.image.tag }}
+{{- $pullPolicy := .Values.image.pullPolicy }}
+
+{{ range .Values.additionalDeployments }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: "{{ $fullname }}-{{ .name }}"
+  labels:
+    app.kubernetes.io/name: {{ $name }}
+    helm.sh/chart: {{ $chart }}
+    app.kubernetes.io/instance: {{ $relName }}
+    app.kubernetes.io/managed-by: {{ $relService }}
+spec:
+  replicas: {{ .replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ $name }}
+      app.kubernetes.io/instance: {{ $relName }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ $name }}
+        app.kubernetes.io/instance: {{ $relName }}
+    spec:
+      priorityClassName: {{ $priorityClassName }}
+      containers:
+        - name: {{ $chartName }}
+          image: "{{ $repository }}:{{ $imageTag }}"
+          imagePullPolicy: {{ $pullPolicy }}
+          resources:
+            {{- toYaml .resources | nindent 12 }}
+    {{- with .nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```

Comment:
You need to add `---` at before the last `{{- end }} otherwise it will not create multiple resources when doing the loop as Kubernetes expects this as end of the resource definition.

I tested this locally. Other than that, everything works fine and I'm happy to approve. It sounds weird to me your approach with cluster-overprovisioner by running multiple deployments but since it will be come an optional thing,  I don't see the reason to not have it.

Reply:
ahhh yes, thanks. totally forgot this. fix it.

we had multiple node pools with different cpu/memory setups for different application with different requirements.